back/app/policies/stat_comment_policy.rb

```diff
@@ -9,78 +9,87 @@ def initialize(user, scope)
     end
 
     def resolve
-      if user&.active? && user.admin?
-        scope.all
-      elsif user&.active? && user.project_moderator?
-        projects = ProjectPolicy::Scope.new(user, Project.all).resolve
-        # we're deliberately avoiding to join ideas to the main scope itself,
-        # because it conflicts with other queries modifying the scope (e.g.
-        # filtering on projects)
-        scope.where(post_type: 'Idea', post_id: Idea.where(project: projects))
-      else
-        scope.none
-      end
+      return scope.none unless user&.active?
+
+      resolve_for_active
+    end
+
+    def resolve_for_active
+      user.admin? ? scope.all : scope.none
     end
```

Comment:
also a recurring pattern: we create an helper method without the guard, to make the patching easier. Otherwise we would have to repeat the guard in some cases in the patch.